expr/func_aspercent_test.go

```diff
@@ -0,0 +1,649 @@
+				QueryPatt:  "func(tag=something;tag2=anything)",
+				Target:     "a;tag=something;tag2=anything",
+				Datapoints: getCopy(a),
+			},
+		},
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "asPercent(func(tag=something;tag2=anything),sumSeries(func(tag=something;tag2=anything)))",
+				Target:     "asPercent(a;tag=something;tag2=anything,sumSeries(func(tag=something;tag2=anything)))",
+				Datapoints: out,
+			},
+		},
+		t,
+		math.NaN(),
+		nil,
+		nil,
+		"None",
+	)
+}
+
+func TestAsPercentMultipleNoArg(t *testing.T) {
+	out1 := []schema.Point{
+		{Val: math.NaN(), Ts: 10},
+		{Val: 0, Ts: 20},
+		{Val: 5.5 / (math.MaxFloat64 - 14.5) * 100, Ts: 30},
+		{Val: math.NaN(), Ts: 40},
+		{Val: math.NaN(), Ts: 50},
+		{Val: 100, Ts: 60},
+	}
+	out2 := []schema.Point{
+		{Val: math.NaN(), Ts: 10},
+		{Val: 100, Ts: 20},
+		{Val: (math.MaxFloat64 - 20) / (math.MaxFloat64 - 14.5) * 100, Ts: 30},
+		{Val: math.NaN(), Ts: 40},
+		{Val: 100, Ts: 50},
+		{Val: math.NaN(), Ts: 60},
+	}
+	testAsPercent(
+		"multi-noarg",
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "func(tag=something;tag2=anything)",
+				Target:     "a;tag=something;tag2=anything",
+				Datapoints: getCopy(a),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  "func(tag=something;tag2=anything)",
+				Target:     "b;tag=something;tag2=anything",
+				Datapoints: getCopy(b),
+			},
+		},
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "asPercent(func(tag=something;tag2=anything),sumSeries(func(tag=something;tag2=anything)))",
+				Target:     "asPercent(a;tag=something;tag2=anything,sumSeries(func(tag=something;tag2=anything)))",
+				Datapoints: out1,
+			},
+			{
+				Interval:   10,
+				QueryPatt:  "asPercent(func(tag=something;tag2=anything),sumSeries(func(tag=something;tag2=anything)))",
+				Target:     "asPercent(b;tag=something;tag2=anything,sumSeries(func(tag=something;tag2=anything)))",
+				Datapoints: out2,
+			},
+		},
+		t,
+		math.NaN(),
+		nil,
+		nil,
+		"None",
+	)
+}
+
+func TestAsPercentTotalFloat(t *testing.T) {
+	total := 123.456
+	out := []schema.Point{
+		{Val: 0, Ts: 10},
+		{Val: 0, Ts: 20},
+		{Val: 5.5 / total * 100, Ts: 30},
+		{Val: math.NaN(), Ts: 40},
+		{Val: math.NaN(), Ts: 50},
+		{Val: 1234567890 / total * 100, Ts: 60},
+	}
+	testAsPercent(
+		"single-noarg",
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "func(a;tag=something;tag2=anything)",
+				Target:     "a;tag=something;tag2=anything",
+				Datapoints: getCopy(a),
+			},
+		},
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "asPercent(func(a;tag=something;tag2=anything),123.456)",
+				Target:     "asPercent(a;tag=something;tag2=anything,123.456)",
+				Datapoints: out,
+			},
+		},
+		t,
+		total,
+		nil,
+		nil,
+		"123.456",
+	)
+}
+
+func TestAsPercentTotalSerie(t *testing.T) {
+	out1 := []schema.Point{
+		{Val: math.NaN(), Ts: 10},
+		{Val: math.NaN(), Ts: 20},
+		{Val: math.Inf(0), Ts: 30},
```

Comment:
hmm? b/a = `(math.MaxFloat64 - 20) / 5.5`, what am i missing here?
anyway as i suggested yesterday I think the tests should be rewritten so that the failures are within the Test functions and not within the testAsPercent helper.

Reply:
* 100 -> overflow